mne/io/base.py

```diff
@@ -836,6 +836,102 @@ def __setitem__(self, item, value):
         # set the data
         self._data[sel, start:stop] = value
 
+    def get_data(self, picks=None, start=0, stop=None,
+                 reject_by_annotation=None, return_times=False):
+        """Get data in the given range.
+
+        Parameters
+        ----------
+        picks : array-like of int | None
+            Indices of channels to get data from. If None, data from all
+            channels is returned
+        start : int
+            The first sample to include. Defaults to 0.
+        stop : int | None
+            End sample (first not to include). If None (default), the end of
+            the data is  used.
+        reject_by_annotation : None | 'omit' | 'NaN'
+            Whether to reject by annotation. If None (default), no rejection is
+            done. If 'omit', segments annotated with description starting with
+            'bad' are omitted. If 'NaN', the bad samples are filled with NaNs.
+        return_times : bool
+            Whether to return times as well.
+
+        Returns
+        -------
+        data : ndarray, shape (n_channels, n_times)
+            Copy of the data in the given range.
+        times : ndarray, shape (n_times,)
+            Times associated with the data samples. Only returned if
+            return_times=True.
+
+        Notes
+        -----
+
+        .. versionadded:: 0.14.0
+        """
+        if picks is None:
+            picks = range(self.info['nchan'])
+        if self.annotations is None or reject_by_annotation is None:
+            data, times = self[picks, start:stop]
+            if return_times:
+                return data, times
+            return data
+        if reject_by_annotation not in ['omit', 'NaN']:
```

Comment:
`'nan'` should probably be okay, too, it's how NumPy prints it
```
array([ nan,  nan,  nan])
```
so you can just use `.lower`